<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Keita Iida (M. Haga, K. Iida, M. Okada)" />

<meta name="date" content="2023-09-29" />

<title>Statistical analysis for the Venn diagram (Figs. 1G and H)</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Statistical analysis for the Venn diagram
(Figs. 1G and H)</h1>
<h4 class="author">Keita Iida (M. Haga, K. Iida, M. Okada)</h4>
<h4 class="date">2023-09-29</h4>


<div id="TOC">
<ul>
<li><a href="#load-the-data" id="toc-load-the-data"><span class="toc-section-number">1</span> Load the data</a>
<ul>
<li><a href="#all-genes-in-figure-1g" id="toc-all-genes-in-figure-1g"><span class="toc-section-number">1.1</span> All genes in Figure 1G</a></li>
<li><a href="#differentially-expressed-genes-in-figure-1g" id="toc-differentially-expressed-genes-in-figure-1g"><span class="toc-section-number">1.2</span> Differentially expressed genes in
Figure 1G</a></li>
<li><a href="#all-genes-in-figure-1h" id="toc-all-genes-in-figure-1h"><span class="toc-section-number">1.3</span> All genes in Figure 1H</a></li>
<li><a href="#differentially-downregulated-genes-in-figure-1h" id="toc-differentially-downregulated-genes-in-figure-1h"><span class="toc-section-number">1.4</span> Differentially downregulated genes
in Figure 1H</a></li>
<li><a href="#differentially-upregulated-genes-in-figure-1h" id="toc-differentially-upregulated-genes-in-figure-1h"><span class="toc-section-number">1.5</span> Differentially upregulated genes
in Figure 1H</a></li>
</ul></li>
<li><a href="#statistical-test" id="toc-statistical-test"><span class="toc-section-number">2</span> Statistical test</a>
<ul>
<li><a href="#figure-1g" id="toc-figure-1g"><span class="toc-section-number">2.1</span> Figure 1G</a></li>
<li><a href="#figure-1h-downregulated" id="toc-figure-1h-downregulated"><span class="toc-section-number">2.2</span> Figure 1H (downregulated)</a></li>
<li><a href="#figure-1h-upregulated" id="toc-figure-1h-upregulated"><span class="toc-section-number">2.3</span> Figure 1H (upregulated)</a></li>
</ul></li>
</ul>
</div>

<p><br></p>
<div id="load-the-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Load the data</h1>
<div id="all-genes-in-figure-1g" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> All genes in Figure
1G</h2>
<p>Load the data</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_G_RNAseq_gene.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig_1G_ATAC_&amp;_ChIP_gene_list.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>all_fig1g <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">rna =</span> <span class="fu">as.character</span>(data1[, <span class="dv">1</span>]),</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                  <span class="at">atac =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]),</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>                  <span class="at">chip =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]))</span></code></pre></div>
</div>
<div id="differentially-expressed-genes-in-figure-1g" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Differentially
expressed genes in Figure 1G</h2>
<p>Load the data</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_G_RNAseq_gene.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_G_ATACseq_gene.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>data3 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_G_ChIPseq_gene.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>deg_fig1g <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">rna =</span> <span class="fu">as.character</span>(data1[, <span class="dv">1</span>]),</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>                  <span class="at">atac =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]),</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>                  <span class="at">chip =</span> <span class="fu">as.character</span>(data3[, <span class="dv">1</span>]))</span></code></pre></div>
</div>
<div id="all-genes-in-figure-1h" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> All genes in Figure
1H</h2>
<p>Load the data</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vivo_gene_list.csv&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vitro_gene_list.csv&quot;</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>all_fig1h <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vivo =</span> <span class="fu">as.character</span>(data1[, <span class="dv">1</span>]),</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>                  <span class="at">vitro =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]))</span></code></pre></div>
</div>
<div id="differentially-downregulated-genes-in-figure-1h" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Differentially
downregulated genes in Figure 1H</h2>
<p>Load the data</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vivo_downregulate.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vitro_downregulate.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>deg_down_fig1h <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vivo =</span> <span class="fu">as.character</span>(data1[, <span class="dv">1</span>]),</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>                       <span class="at">vitro =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]))</span></code></pre></div>
</div>
<div id="differentially-upregulated-genes-in-figure-1h" class="section level2" number="1.5">
<h2><span class="header-section-number">1.5</span> Differentially
upregulated genes in Figure 1H</h2>
<p>Load the data</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>data1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vivo_upregulate.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;data/Fig1_H_in_vitro_upregulate.csv&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>deg_up_fig1h <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vivo =</span> <span class="fu">as.character</span>(data1[, <span class="dv">1</span>]),</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>                     <span class="at">vitro =</span> <span class="fu">as.character</span>(data2[, <span class="dv">1</span>]))</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="statistical-test" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Statistical test</h1>
<div id="figure-1g" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Figure 1G</h2>
<p>Obtain the genes common to RNA-seq, ATAC-seq, and ChIP-seq
datasets.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>deg_common <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(deg_fig1g<span class="sc">$</span>rna, deg_fig1g<span class="sc">$</span>atac,</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>                                     deg_fig1g<span class="sc">$</span>chip))</span></code></pre></div>
<p>Perform statistical tests by the following procedure: 1. Randomly
select the genes from each of <code>all_fig1g$rna</code>,
<code>all_fig1g$atac</code> and <code>all_fig1g$chip</code>, in which
the number of genes is the same for <code>deg_fig1g$rna</code>,
<code>deg_fig1g$atac</code>, and <code>deg_fig1g$chip</code>,
respectively. 2. Take an intersection among the selected gene sets. 3.
Perform (1) and (2) 100000 times and compute a probability (i.e.,
p-value) that the number of genes obtained in (2) is greater than that
of <code>deg_common</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_fig1g<span class="sc">$</span>rna)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_fig1g<span class="sc">$</span>atac)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>l <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_fig1g<span class="sc">$</span>chip)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>ngenes_common <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="dv">100000</span>)){</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  genes_rna <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1g<span class="sc">$</span>rna, <span class="at">size =</span> n, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>  genes_atac <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1g<span class="sc">$</span>atac, <span class="at">size =</span> m, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  genes_chip <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1g<span class="sc">$</span>chip, <span class="at">size =</span> l, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  ngenes_common <span class="ot">&lt;-</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>    <span class="fu">c</span>(ngenes_common,</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>      <span class="fu">length</span>(<span class="fu">intersect</span>(<span class="fu">intersect</span>(genes_rna, genes_atac), genes_chip)))</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>}</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">sum</span>(ngenes_common <span class="sc">&gt;</span> <span class="fu">length</span>(deg_common)))</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<p>This means that the probability is less than 0.001 % (or p-value &lt;
1e-05).</p>
<p><br></p>
</div>
<div id="figure-1h-downregulated" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Figure 1H
(downregulated)</h2>
<p>Obtain the genes common to vivo and vitro datasets.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>deg_common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(deg_down_fig1h<span class="sc">$</span>vivo, deg_down_fig1h<span class="sc">$</span>vitro)</span></code></pre></div>
<p>Perform statistical tests by the following procedure: 1. Randomly
select the genes from each of <code>all_fig1h$vivo</code> and
<code>all_fig1h$vitro</code>, in which the number of genes is the same
for <code>deg_down_fig1h$vivo</code> and
<code>deg_down_fig1h$vitro</code>, respectively. 2. Take an intersection
among the selected gene sets. 3. Perform (1) and (2) 100000 times and
compute a probability (i.e., p-value) that the number of genes obtained
in (2) is greater than that of <code>deg_common</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_down_fig1h<span class="sc">$</span>vivo)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_down_fig1h<span class="sc">$</span>vitro)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>ngenes_common <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="dv">100000</span>)){</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  genes_vivo <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1h<span class="sc">$</span>vivo, <span class="at">size =</span> n, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  genes_vitro <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1h<span class="sc">$</span>vitro, <span class="at">size =</span> m, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  ngenes_common <span class="ot">&lt;-</span> <span class="fu">c</span>(ngenes_common, <span class="fu">length</span>(<span class="fu">intersect</span>(genes_vivo, genes_vitro)))</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>}</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">sum</span>(ngenes_common <span class="sc">&gt;</span> <span class="fu">length</span>(deg_common)))</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<p>This means that the probability is less than 0.001 % (or p-value &lt;
1e-05).</p>
<p><br></p>
</div>
<div id="figure-1h-upregulated" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Figure 1H
(upregulated)</h2>
<p>Obtain the genes common to RNA-seq, ATAC-seq, and ChIP-seq
datasets.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>deg_common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(deg_up_fig1h<span class="sc">$</span>vivo, deg_up_fig1h<span class="sc">$</span>vitro)</span></code></pre></div>
<p>Perform statistical tests by the following procedure: 1. Randomly
select the genes from each of <code>all_fig1h$vivo</code> and
<code>all_fig1h$vitro</code>, in which the number of genes is the same
for <code>deg_up_fig1h$vivo</code> and <code>deg_up_fig1h$vitro</code>,
respectively. 2. Take an intersection among the selected gene sets. 3.
Perform (1) and (2) 100000 times and compute a probability (i.e.,
p-value) that the number of genes obtained in (2) is greater than that
of <code>deg_common</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_up_fig1h<span class="sc">$</span>vivo)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">length</span>(deg_up_fig1h<span class="sc">$</span>vitro)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>ngenes_common <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="dv">100000</span>)){</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>  genes_vivo <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1h<span class="sc">$</span>vivo, <span class="at">size =</span> n, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  genes_vitro <span class="ot">&lt;-</span> <span class="fu">sample</span>(all_fig1h<span class="sc">$</span>vitro, <span class="at">size =</span> m, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a>  ngenes_common <span class="ot">&lt;-</span> <span class="fu">c</span>(ngenes_common, <span class="fu">length</span>(<span class="fu">intersect</span>(genes_vivo, genes_vitro)))</span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a>}</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">sum</span>(ngenes_common <span class="sc">&gt;</span> <span class="fu">length</span>(deg_common)))</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<p>This means that the probability is less than 0.001 % (or p-value &lt;
1e-05).</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
