---
title: "Statistical analysis for the Venn diagram (Figs. 1G and H)"
author: "Keita Iida (M. Haga, K. Iida, M. Okada)"
date: "`r Sys.Date()`"
output:
#  rmarkdown::github_document:
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 2
    number_section: true
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```



<br>

# Load the data
## All genes in Figure 1G
Load the data
```{r, eval = FALSE}
data1 <- read.csv(file = "data/Fig1_G_RNAseq_gene.csv", row.names = 1)
data2 <- read.csv(file = "data/Fig_1G_ATAC_&_ChIP_gene_list.csv", row.names = 1)
all_fig1g <- list(rna = as.character(data1[, 1]),
                  atac = as.character(data2[, 1]),
                  chip = as.character(data2[, 1]))
```

## Differentially expressed genes in Figure 1G
Load the data
```{r, eval = FALSE}
data1 <- read.csv(file = "data/Fig1_G_RNAseq_gene.csv", row.names = 1)
data2 <- read.csv(file = "data/Fig1_G_ATACseq_gene.csv", row.names = 1)
data3 <- read.csv(file = "data/Fig1_G_ChIPseq_gene.csv", row.names = 1)
deg_fig1g <- list(rna = as.character(data1[, 1]),
                  atac = as.character(data2[, 1]),
                  chip = as.character(data3[, 1]))
```

## All genes in Figure 1H
Load the data
```{r, eval = FALSE}
data1 <- read.csv(file = "data/Fig1_H_in_vivo_gene_list.csv")
data2 <- read.csv(file = "data/Fig1_H_in_vitro_gene_list.csv")
all_fig1h <- list(vivo = as.character(data1[, 1]),
                  vitro = as.character(data2[, 1]))
```

## Differentially downregulated genes in Figure 1H
Load the data
```{r, eval = FALSE}
data1 <- read.csv(file = "data/Fig1_H_in_vivo_downregulate.csv", row.names = 1)
data2 <- read.csv(file = "data/Fig1_H_in_vitro_downregulate.csv", row.names = 1)
deg_down_fig1h <- list(vivo = as.character(data1[, 1]),
                       vitro = as.character(data2[, 1]))
```

## Differentially upregulated genes in Figure 1H
Load the data
```{r, eval = FALSE}
data1 <- read.csv(file = "data/Fig1_H_in_vivo_upregulate.csv", row.names = 1)
data2 <- read.csv(file = "data/Fig1_H_in_vitro_upregulate.csv", row.names = 1)
deg_up_fig1h <- list(vivo = as.character(data1[, 1]),
                     vitro = as.character(data2[, 1]))
```



<br>

# Statistical test
## Figure 1G
Obtain the genes common to RNA-seq, ATAC-seq, and ChIP-seq datasets.
```{r, eval = FALSE}
deg_common <- Reduce(intersect, list(deg_fig1g$rna, deg_fig1g$atac,
                                     deg_fig1g$chip))
```

Perform statistical tests by the following procedure:
1. Randomly select the genes from each of `all_fig1g$rna`, `all_fig1g$atac` and
`all_fig1g$chip`, in which the number of genes is the same for
`deg_fig1g$rna`, `deg_fig1g$atac`, and `deg_fig1g$chip`, respectively.
2. Take an intersection among the selected gene sets.
3. Perform (1) and (2) 100000 times and compute a probability (i.e., p-value)
that the number of genes obtained in (2) is greater than that of `deg_common`.
```{r, eval = FALSE}
n <- length(deg_fig1g$rna)
m <- length(deg_fig1g$atac)
l <- length(deg_fig1g$chip)
ngenes_common <- c()
set.seed(1)
for(i in seq_len(100000)){
  genes_rna <- sample(all_fig1g$rna, size = n, replace = FALSE, prob = NULL)
  genes_atac <- sample(all_fig1g$atac, size = m, replace = FALSE, prob = NULL)
  genes_chip <- sample(all_fig1g$chip, size = l, replace = FALSE, prob = NULL)
  ngenes_common <-
    c(ngenes_common,
      length(intersect(intersect(genes_rna, genes_atac), genes_chip)))
}
print(sum(ngenes_common > length(deg_common)))
```

```
[1] 0
```

This means that the probability is less than 0.001 % (or p-value < 1e-05).



<br>

## Figure 1H (downregulated)
Obtain the genes common to vivo and vitro datasets.
```{r, eval = FALSE}
deg_common <- intersect(deg_down_fig1h$vivo, deg_down_fig1h$vitro)
```

Perform statistical tests by the following procedure:
1. Randomly select the genes from each of `all_fig1h$vivo` and
`all_fig1h$vitro`, in which the number of genes is the same for
`deg_down_fig1h$vivo` and `deg_down_fig1h$vitro`, respectively.
2. Take an intersection among the selected gene sets.
3. Perform (1) and (2) 100000 times and compute a probability (i.e., p-value)
that the number of genes obtained in (2) is greater than that of `deg_common`.
```{r, eval = FALSE}
n <- length(deg_down_fig1h$vivo)
m <- length(deg_down_fig1h$vitro)
ngenes_common <- c()
set.seed(1)
for(i in seq_len(100000)){
  genes_vivo <- sample(all_fig1h$vivo, size = n, replace = FALSE, prob = NULL)
  genes_vitro <- sample(all_fig1h$vitro, size = m, replace = FALSE, prob = NULL)
  ngenes_common <- c(ngenes_common, length(intersect(genes_vivo, genes_vitro)))
}
print(sum(ngenes_common > length(deg_common)))
```

```
[1] 0
```

This means that the probability is less than 0.001 % (or p-value < 1e-05).



<br>

## Figure 1H (upregulated)
Obtain the genes common to RNA-seq, ATAC-seq, and ChIP-seq datasets.
```{r, eval = FALSE}
deg_common <- intersect(deg_up_fig1h$vivo, deg_up_fig1h$vitro)
```

Perform statistical tests by the following procedure:
1. Randomly select the genes from each of `all_fig1h$vivo` and
`all_fig1h$vitro`, in which the number of genes is the same for
`deg_up_fig1h$vivo` and `deg_up_fig1h$vitro`, respectively.
2. Take an intersection among the selected gene sets.
3. Perform (1) and (2) 100000 times and compute a probability (i.e., p-value)
that the number of genes obtained in (2) is greater than that of `deg_common`.
```{r, eval = FALSE}
n <- length(deg_up_fig1h$vivo)
m <- length(deg_up_fig1h$vitro)
ngenes_common <- c()
set.seed(1)
for(i in seq_len(100000)){
  genes_vivo <- sample(all_fig1h$vivo, size = n, replace = FALSE, prob = NULL)
  genes_vitro <- sample(all_fig1h$vitro, size = m, replace = FALSE, prob = NULL)
  ngenes_common <- c(ngenes_common, length(intersect(genes_vivo, genes_vitro)))
}
print(sum(ngenes_common > length(deg_common)))
```

```
[1] 0
```

This means that the probability is less than 0.001 % (or p-value < 1e-05).
